File: backend/src/schemas/permissions.py
from pydantic import BaseModel
from typing import Literal

class PermissionAddRequest(BaseModel):
    target_type : Literal["branch", "folder"]
    target_identifier : str
    allow_write : bool

class PermissionResponse(BaseModel):
    id : int
    repository_id : int
    team_id : int
    team_name : str
    target_type : Literal["branch", "folder"]
    target_identifier : str
    allow_write : bool

File: frontend/src/lib/schemas/permissions.ts
import { z } from "zod"

export const permissionSchema = z.object({
  team_id: z.number().int().positive("Team is required"),
  target_type: z.enum(["branch", "folder"]),
  target_identifier: z.string().min(1, "Branch name is required").max(255),
  allow_write: z.boolean(),
})

export type PermissionInput = z.infer<typeof permissionSchema>
